test/unit/ready.js

```diff
@@ -67,4 +91,32 @@ QUnit.module( "ready" );
 			"Argument passed to fn in jQuery(document).ready( fn ) should be jQuery" );
 	} );
 
+	QUnit.test( "Promise.resolve(jQuery.ready)", function( assert ) {
+		assert.expect( 2 );
+		var done = jQuery.map( new Array( 2 ), function() { return assert.async(); } );
```

Comment:
@Krinkle `assert.async` now accepts a number specifying how many times it should get called so this is sort of blessed by upstream currently... We planned to switch to this pattern once we update QUnit.